qcodes/instrument_drivers/tektronix/TPS2012.py

```diff
@@ -0,0 +1,243 @@
+              byte_order, no_of_points, waveform_ID, point_format,
+              x_incr, x_zero, x_unit, y_multiplier, y_zero, y_offset, y_unit
+        """
+        response = response.split(';')
+        outdict = {}
+        outdict['no_of_bytes'] = response[0]
+        outdict['no_of_bits'] = response[1]
+        outdict['encoding'] = response[2]
+        outdict['binary_format'] = response[3]
+        outdict['byte_order'] = response[4]
+        outdict['no_of_points'] = response[5]
+        outdict['waveform_ID'] = response[6]
+        outdict['point_format'] = response[7]
+        outdict['x_incr'] = response[8]
+        # outdict['point_offset'] = response[9]  # Always zero
+        outdict['x_zero'] = response[10]
+        outdict['x_unit'] = response[11]
+        outdict['y_multiplier'] = response[12]
+        outdict['y_zero'] = response[13]
+        outdict['y_offset'] = response[14]
+        outdict['y_unit'] = response[15]
+
+        return outdict
+
+    def _curveparameterparser(self, waveform):
+        """
+        The parser for the curve parameter. Note that WAVFrm? is equivalent
+        to WFMPre?; CURVe?
+
+        Args:
+            waveform (str): The return value of WAVFrm?
+
+        Returns:
+            (np.array, np.array): Two numpy arrays with the time axis in units
+            of s and curve values in units of V; (time, voltages)
+        """
+        fulldata = waveform.split(';')
+        preamblestr = ';'.join(fulldata[:16])
+        curvestr = ';'.join(fulldata[16:])
+
+        preamble = self._preambleparser(preamblestr)
+        curvestr = curvestr[2+len(preamble['no_of_bytes']) *
+                            len(preamble['no_of_points']):-1]
+        rawcurve = self._binaryparser(curvestr)
+
+        yoff = float(preamble['y_offset'])
+        yoff -= 2**15  # data width 2
+        ymult = float(preamble['y_multiplier'])
+        ydata = ymult*(rawcurve-yoff)
+
+        xstart = float(preamble['x_zero'])
+        xinc = float(preamble['x_incr'])
+        xdata = np.linspace(xstart, len(ydata)*xinc+xstart, len(ydata))
+        return (xdata, ydata)
+
+    ##################################################
+    # METHODS FOR THE USER                           #
+    ##################################################
+
+    def clear_message_queue(self, verbose=False):
+        """
+        Function to clear up (flush) the VISA message queue of the AWG
+        instrument. Reads all messages in the the queue.
+
+        Args:
+            verbose (Bool): If True, the read messages are printed.
+                Default: False.
+        """
+        original_timeout = self.visa_handle.timeout
+        self.visa_handle.timeout = 1000  # 1 second as VISA counts in ms
+        gotexception = False
+        while not gotexception:
+            try:
+                message = self.visa_handle.read()
+                if verbose:
+                    print(message)
```

Comment:
No, I think printing is better. We talked about this already when I put the exact same function in the AWG driver (but actually, I think this function should be promoted to be in the VisaInstrument class), but in this case, verbosity is really for the user to read in his/her terminal what was still in the queue.